### PR TITLE
Fix plug-in peerDep check

### PR DESCRIPTION
**Pull request by zFernand0**
**What It Does**

Remove the warnings at plug-in install time that appear, even though the plug-in peerDep version-range satisfies the requirement.

**How to Test**

First you would want to reproduce the issue:
- Globally install any of the CLI versions tagged as 7.0.0-next.###
  - https://github.com/zowe/zowe-cli/tags?after=v7.0.0
- Modify a plug-in's peerDep version of imperative to `>=5.0.0-next.0 <5.0.0`.
- Install your plug-in
- Notice that there is a warning at install time

Then you can test this branch:
- Build this branch and install/link it globally
- Update the plug-in peerDep version to `>=8.0.0-next.0 <8.0.0`
- Install the plug-in
- Notice that there is no warning at install time 😋 

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

This follows the same behavior as https://semver.npmjs.com/
- Feel free to go to the website and enter the values `@zowe/imperative` and `>=5.0.0-next.0 <5.0.0` 😋 

Also, this GIF might make things easier to review:
temp

### CODE REVIEW

**Review by awharn — APPROVED** [review]

File: packages/imperative/src/imperative/src/plugins/PluginManagementFacility.ts
Comment (traeok) [inline packages/imperative/src/imperative/src/plugins/PluginManagementFacility.ts:523]:
```suggestion
     * If the versions do not satisfy the semver rules, then a
```

File: packages/imperative/src/imperative/src/plugins/PluginManagementFacility.ts
Comment (traeok) [inline packages/imperative/src/imperative/src/plugins/PluginManagementFacility.ts:934]: Thoughts on changing line 934 to "We consider the versions to be compatible if they satisfy the CLI version range." ?

**Review by traeok — COMMENTED** [review]
Had a couple suggestions about phrasing in typedoc 🙂

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @zFernand0!

**Review by traeok — APPROVED** [review]
LGTM - thanks @zFernand0!

**Review by gejohnston — APPROVED** [review]
My only comment is on the example in the PR description.

A range of >=5.0.0-next.0 <5.0.0 would only permit the use pre-release versions. I think it would fail when that version becomes GA (5.0.0).

I think in practice, that a range of >=5.0.0-next.0 <6.0.0 would permit pre-release versions of 5.0.0, and permit every GA version of 5.x, but not permit any 6.x version.

**Comment by zFernand0** [comment]
This PR can wait until after to following PR is merged into `next` to avoid any confusion/conflicts 😋 
- https://github.com/zowe/zowe-cli/pull/2073

**Pull request merged**
